src/Adapter/Product/Update/ProductShopUpdater.php

```diff
@@ -0,0 +1,306 @@
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+namespace PrestaShop\PrestaShop\Adapter\Product\Update;
+
+use DateTime;
+use Doctrine\DBAL\Connection;
+use Doctrine\DBAL\Query\QueryBuilder;
+use PrestaShop\PrestaShop\Adapter\Product\Repository\ProductRepository;
+use PrestaShop\PrestaShop\Adapter\Shop\Repository\ShopRepository;
+use PrestaShop\PrestaShop\Core\Domain\Product\ValueObject\ProductId;
+use PrestaShop\PrestaShop\Core\Domain\Shop\ValueObject\ShopId;
+use PrestaShop\PrestaShop\Core\Util\DateTime\DateTime as DateTimeUtil;
+use Product;
+
+class ProductShopUpdater
+{
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    /**
+     * @var string
+     */
+    private $dbPrefix;
+
+    /**
+     * @var ProductRepository
+     */
+    private $productRepository;
+
+    /**
+     * @var ShopRepository
+     */
+    private $shopRepository;
+
+    /**
+     * @param Connection $connection
+     * @param string $dbPrefix
+     * @param ProductRepository $productRepository
+     * @param ShopRepository $shopRepository
+     */
+    public function __construct(
+        Connection $connection,
+        string $dbPrefix,
+        ProductRepository $productRepository,
+        ShopRepository $shopRepository
+    ) {
+        $this->connection = $connection;
+        $this->dbPrefix = $dbPrefix;
+        $this->productRepository = $productRepository;
+        $this->shopRepository = $shopRepository;
+    }
+
+    /**
+     * @param ProductId $productId
+     * @param ShopId $sourceShopId
+     * @param ShopId $targetShopId
+     */
+    public function copyToShop(ProductId $productId, ShopId $sourceShopId, ShopId $targetShopId): void
+    {
+        $this->shopRepository->assertShopExists($sourceShopId);
+        $this->shopRepository->assertShopExists($targetShopId);
+
+        /** @var Product $sourceProduct */
+        $sourceProduct = $this->productRepository->get($productId, $sourceShopId);
+
+        // @todo: for now only fields from product_shop table ar handed, we still need to handle multilang, stock_available, and probably other things (in another PR)
+        $fields = [];
+        // The fields are fetched separately for more clarity, and it could also allow to configure which parts are copied
+        // (e.g copy only prices but not stock)
+        $fields = $fields + $this->getAssociationFields($sourceProduct);
+        $fields = $fields + $this->getPricesFields($sourceProduct);
+        $fields = $fields + $this->getStockFields($sourceProduct);
+        $fields = $fields + $this->getSEOFields($sourceProduct);
+        $fields = $fields + $this->getOptionsFields($sourceProduct);
+        $fields = $fields + $this->getCustomizationFields($sourceProduct);
+        $fields = $fields + $this->getDateFields($sourceProduct);
```

Comment:
IMO this will be easier to read (and less risky) if you use a single array_merge operation
```suggestion
        // The fields are fetched separately for more clarity, and it could also allow to configure which parts are copied
        // (e.g copy only prices but not stock)
        $fields = array_merge(
            $this->getAssociationFields($sourceProduct),
            $this->getPricesFields($sourceProduct),
            $this->getStockFields($sourceProduct),
            $this->getSEOFields($sourceProduct),
            $this->getOptionsFields($sourceProduct),
            $this->getCustomizationFields($sourceProduct),
            $this->getDateFields($sourceProduct)
        );
```